### PR TITLE
Enable user to clear console history with a click (#1015)

### PR DESCRIPTION
**Pull request by imranariffin**
### Pull Request checklist

- [ ] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](https://github.com/iodide-project/iodide/blob/master/CHANGELOG.md) with any user-visible changes.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

This PR should implement feature #1015 
* Add a clear history item to the list of drop down next to console input
* Add separator line between clear console action and language selection in the drop-down
* Clear the console history on clicking the clear console drop-down item

### Screenshots: Before Change

Screenshot from 2019-11-19 00-08-05

#### Before clear history

Screenshot from 2019-11-18 23-25-36

#### After clear history

Screenshot from 2019-11-18 23-25-14

### CODE REVIEW

**Comment by imranariffin** [comment]
Seems like I got 
```
HTTP 400
Commit sha does not match Circle build.
Exited with code 1
```
in both of the failing tests. What does this mean?

File: src/editor/console/input/components/__tests__/console-language-menu.test.js [hunk 1]
```diff
@@ -26,4 +37,41 @@ describe("ConsoleLanguageMenu mapStateToProps", () => {
...
+  });
+  it("sets shouldDisplayClearConsoleAction prop as false when history empty", () => {
+    state.history = [];
+    expect(mapStateToProps(state).shouldDisplayClearConsoleAction).toEqual(
+      false
+    );
+  });
+});
+
+describe("ConsoleLanguageMenu mapStateToProps", () => {
```
Comment (imranariffin) [inline src/editor/console/input/components/__tests__/console-language-menu.test.js]:
```suggestion
describe("ConsoleLanguageMenu clearConsoleHistory", () => {
```

File: src/editor/console/history/__tests__/reducer.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        {
+          content: "1+2",
+          historyId: "7k77hjno1d",
+          historyType: "CONSOLE_INPUT",
+          language: "py"
+        },
+        {
+          historyId: "6316fcv86c",
+          historyType: "CONSOLE_OUTPUT"
+        }
```
Comment (wlach) [inline src/editor/console/history/__tests__/reducer.test.js]: Probably not necessary to have quite this many console entries for a valid test -- just 2 should be sufficient.

Comment (imranariffin) [reply src/editor/console/history/__tests__/reducer.test.js]: I see, agreed that more than two entries is not necessary. I can remove one entry if that's okay post-approval.

Comment (imranariffin) [reply src/editor/console/history/__tests__/reducer.test.js]: 2a60407d

**Comment by bcolloran** [comment]
looks great to me too! thanks @imranariffin! :-)

(and @wlach thanks to you too!)

**Comment by imranariffin** [comment]
Thank you for the reviews @bcolloran & @wlach!

**Pull request merged**